rt/test/TESTOPR2.MLC

```diff
@@ -0,0 +1,1443 @@
+&DCLEN   SETA  DCLEN('&&')               two ampersands
+         MNOTE 'DCLEN(''&&&&'')=&DCLEN'
+         AIF   (&DCLEN NE 1).DCLERR
+&DCLEN   SETA  DCLEN('a''''b')           string is "a'b"
+         MNOTE 'DCLEN(''a''''''''b'')=&DCLEN'
+         AIF   (&DCLEN NE 3).DCLERR
+&DCLEN   SETA  DCLEN('a''''b&&c')        string is "a'b&c""
+         MNOTE 'DCLEN(''a''''''''b&&&&c'')=&DCLEN'
+         AIF   (&DCLEN NE 5).DCLERR
+&DCLEN   SETA  DCLEN('&&&&'.'''''''')    string is "&&''"
+         MNOTE 'DCLEN(''&&&&&&&&''.'''''''''''''''')=&DCLEN'
+         AIF   (&DCLEN NE 4).DCLERR
+*----------------------------------------------------------------------
+*        z390 tests
+
+&DCLSTR  SETC  '''''''''&&&&&&&&1'''''''''  len 17; see MNOTE below
+&K       SETA  K'&DCLSTR 
+&DDCLSTR SETC  DOUBLE('&DCLSTR')            double for MNOTE and DC
+*
+         MNOTE 'K''&&DCLSTR=&K  &&DCLSTR = >&DDCLSTR<'
+***********************************************************************
+*        On z390 the preceding MNOTE displays
+*
+*  MNOTE 'K'&DCLSTR=17  &DCLSTR = >''''''''&&&&&&&&&&&&&&&&1''''''''<'
+*
+*        Using HLASM the MNOTE displays
+*
+* +K'&DCLSTR=17  &DCLSTR = >''''&&&&&&&&1''''<           
+*
+*        Looks like mz390 just shows the MNOTE with the message
+*        text doubled, not the actual output of the MNOTE.
+*
+*        The following DC statement is the same for both z390 & HLASM.
+***********************************************************************
+         PUSH  PRINT
+         PRINT DATA
+         DC    FL1'&K',C'>&DDCLSTR<' 
+         POP   PRINT
+*
+&DCLEN   SETA  DCLEN('&DCLSTR') 
+         MNOTE 'DCLEN(''&&DCLSTR'')=&DCLEN' 
+         AIF   (&DCLEN NE 9).DCLERR
+*----------------------------------------------------------------------
+         MNOTE '********** End   DCLEN tests **********'
+*======================================================================
+         AGO   .SKPDCVO    DCVAL tests now done in rt/mlc/TOPR2.MLC
+         MNOTE '********** Begin DCVAL tests **********'
+*----------------------------------------------------------------------
+*        HLASM LR examples
+*
+******** old DCVAL tests
+&DCVAL   SETC  DCVAL('')                 null string
+&K       SETA  K'&DCVAL
+         MNOTE 'DCVAL('''')=>&DCVAL<  length = &K'
+         AIF   (K'&DCVAL NE 0).DCVERR
+*
+&DCVAL   SETC  DCVAL('''')               one apostrophe
+&K       SETA  K'&DCVAL
+         MNOTE 'DCVAL('''''''')=>&DCVAL&DCVAL<  length = &K'
+&K       SETA  K'&DCVAL
+         MNOTE 'K''&&DCVAL = &K'
+         AIF   ('&DCVAL' NE '''').DCVERR
+*
+&DCVAL   SETC  DCVAL('&&')               one ampersand
+&K       SETA  K'&DCVAL
+&DCVAL2  SETC  '&DCVAL'.'&DCVAL'
+         MNOTE 'DCVAL(''&&&&'')=>&DCVAL2<'
+         AIF   ('&DCVAL2' NE '&&').DCVERR
+*
+&DCVAL   SETC  DCVAL('a''''b')           string is "a'b"
+&K       SETA  K'&DCVAL
+*        MNOTE 'DCVAL(''a''''''''b'')=>&DCVAL<  length = &K'
+         AIF   ('&DCVAL' NE 'a''b').DCVERR
+* AGO .JJG3
```

Comment:
Could you please either document why this test is being skipped, or just remove the code in its entirety?

Reply:
This AGO is withing the above enclosing AGO .SKPDCV0. It was skipped originally because I could not get the code to work with all the doubled ampersands and single quotes. The new TOPR2.MLC code handles this case. Left it in for posterity. It will be deleted if we agree that the entire DCVAL section here is deleted.